### ISSUE TITLE
Modification of a read-only value attempted

### ISSUE THREAD

**Issue by atoomic** [CpanelInc/Test-MockFile#22]
The following one-liner raises a modification of a read-only value
```
$> perl -MTest::MockFile -E 'use File::Find;
my $d = Test::MockFile->dir( q[/dir], [ "file" ] );
File::Find::find( { no_chdir => 1, wanted => sub { 1 } }, "/dir" )'
# [
#   267,
#   '/dir'
# ]
Modification of a read-only value attempted at /usr/local/cpanel/3rdparty/perl/528/lib/perl5/cpanel_lib/Test/MockFile.pm line 1157.
```

seems to come from *CORE::GLOBAL::opendir override
```perl
         # This isn't a real IO::Dir.
         $_[0] = Test::MockFile::DirHandle->new( $abs_path, $mock_dir->{'contents'} );
```

**Comment by toddr** [comment]
ug...
bareword file handles.

Simplified
```
$>perl -MTest::MockFile -E 'my $d = Test::MockFile->dir( "/dir", [qw/file/] ); opendir(DIR, "/dir") or die'                                  
Modification of a read-only value attempted at /usr/local/cpanel/3rdparty/perl/528/lib/perl5/cpanel_lib/Test/MockFile.pm line 1157.
```

**Comment by toddr** [comment]
This was fixed by 6d8aa360265cdc764b3fe5a3f8f16f6958dd542e
